temporal-sdk/src/test/java/io/temporal/worker/CleanWorkerShutdownTest.java

```diff
@@ -214,141 +98,50 @@ public void testShutdownNow() throws ExecutionException, InterruptedException {
-        found = true;
-        Payloads ar = e.getActivityTaskCompletedEventAttributes().getResult();
-        String r =
-            DataConverter.getDefaultInstance()
-                .fromPayloads(0, Optional.of(ar), String.class, String.class);
-        assertEquals("workershutdown", r);
-      }
-    }
-    assertTrue("Contains ActivityTaskCompleted", found);
-    if (useExternalService) {
-      service.shutdownNow();
-      service.awaitTermination(10, TimeUnit.MINUTES);
-    }
-  }
+  public static class TestWorkflowImpl implements TestWorkflow1 {
 
-  public static class TestWorkflowImpl implements TestWorkflowReturnString {
-
-    private final NoArgsReturnsStringActivity activities =
+    private final TestActivity1 activities =
         Workflow.newActivityStub(
-            NoArgsReturnsStringActivity.class,
+            TestActivity1.class,
             ActivityOptions.newBuilder()
                 .setScheduleToCloseTimeout(Duration.ofSeconds(100))
                 .build());
 
     @Override
-    public String execute() {
-      return activities.execute();
+    public String execute(String now) {
+      return activities.execute(now);
     }
   }
 
-  public static class ActivitiesImpl implements NoArgsReturnsStringActivity {
-    private final CompletableFuture<Boolean> started;
-
-    public ActivitiesImpl(CompletableFuture<Boolean> started) {
-      this.started = started;
-    }
-
-    @Override
-    public String execute() {
-      try {
-        started.complete(true);
-        Thread.sleep(1000);
-      } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
-        return "interrupted";
-      }
-      return "completed";
-    }
-  }
+  public static class ActivitiesImpl implements TestActivity1 {
 
-  public static class HeartbeatingActivitiesImpl implements NoArgsReturnsStringActivity {
-    private final CompletableFuture<Boolean> started;
+    private final CountDownLatch shutdownLatch;
+    private final CountDownLatch shutdownNowLatch;
 
-    public HeartbeatingActivitiesImpl(CompletableFuture<Boolean> started) {
-      this.started = started;
+    public ActivitiesImpl(CountDownLatch shutdownLatch, CountDownLatch shutdownNowLatch) {
+      this.shutdownLatch = shutdownLatch;
+      this.shutdownNowLatch = shutdownNowLatch;
     }
 
     @Override
-    public String execute() {
+    public String execute(String now) {
       try {
-        started.complete(true);
-        Thread.sleep(1000);
-        Activity.getExecutionContext().heartbeat("foo");
-      } catch (ActivityWorkerShutdownException e) {
-        return "workershutdown";
+        if (now == null) {
+          shutdownLatch.countDown();
+        } else {
+          shutdownNowLatch.countDown();
+        }
+        Thread.sleep(10000);
       } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
-        return "interrupted";
+        return INTERRUPTED;
```

Comment:
Why? "Before" we have correct handling of an interrupted exception. "After" - is not correct.
If we have to implement handling in an incorrect way to make things work properly - it's a reason to investigate deeper, not replacing correct handling with incorrect.

Reply:
Interrupted shutdown is tested by making sure the returned value is INTERRUPTED.
The reason we can't interrupt the activity thread is that it seems to prevent gRPC from making the RespondActivityTaskCompleted call to the external service (the RPC call is getting a CANCELLED status code back because the thread is marked as interrupted). This results in the event EVENT_TYPE_ACTIVITY_TASK_COMPLETED missing from history, thus the failing test.

Reply:
> Interrupted shutdown is tested by making sure the returned value is INTERRUPTED.

Reply:
https://github.com/temporalio/sdk-java/issues/731

Reply:
Please add a comment why specifically do we swallow interrupted flag here.
You can add a link to #731 explaining that otherwise the result of the activity is not reported.